iroh-dns-server/src/store/signed_packets.rs

```diff
@@ -110,40 +163,79 @@ impl Actor {
                                 res.send(true).ok();
                             }
                             Message::Remove { key, res } => {
-                                let updated =
-                                    tables.signed_packets.remove(key.as_bytes())?.is_some()
-                                ;
+                                trace!("remove {}", key);
+                                let updated = if let Some(row) = tables.signed_packets.remove(key.as_bytes())? {
+                                    let packet = SignedPacket::from_bytes(&Bytes::copy_from_slice(row.value()))?;
+                                    tables.update_time.remove(&packet.timestamp().to_be_bytes(), key.as_bytes())?;
+                                    inc!(Metrics, store_packets_removed);
+                                    true
+                                } else {
+                                    false
+                                };
                                 if updated {
                                     inc!(Metrics, store_packets_removed);
                                 }
                                 res.send(updated).ok();
                             }
+                            Message::Snapshot { res } => {
+                                trace!("snapshot");
+                                res.send(Snapshot::new(&self.db)?).ok();
+                            }
+                            Message::CheckExpired { key, time } => {
+                                trace!("check expired {} at {}", key, u64::from_be_bytes(time));
+                                tables.update_time.remove_all(&time)?;
```

Comment:
... Should we make this `remove_all` conditional on `time < expired`? 
Also, should we be more precise about which `update_time` record we're removing?

I worry that we're nuking *more* records here than necessary, and as a result, the eviction thread doesn't ask the actor to clean up some keys that were scheduled to be evicted here.

Perhaps ideally this call becomes a `tables.update_time.remove(&time, &key)` call, and it moves into the `if packet.timestamp() < expired` condition.

Reply:
Following up on that even more: Maybe we should abstract out two functions `insert_packet` and `remove_packet` and primarily use those, so we avoid ever forgetting handling both (potentially breaking invariants).

Reply:
> this would only happen for packets that have the exact same time

Yes, that was my thinking, but I don't think this is entirely impossible, and could even happen regularly. It depends on how much traffic there is, the particular pattern that the packets are ingested (bursts?) and potential OS-dependent timer precision.